SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageService.cpp

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialPackageService.h"
+
+#include "Async/Async.h"
+#include "HAL/FileManager.h"
+#include "SpatialGDKServicesConstants.h"
+#include "SpatialGDKServicesModule.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageService);
+
+namespace
+{
+	bool DownloadPackage(const FSpatialPackageDescriptor& Package)
+	{
+		UE_LOG(LogSpatialPackageService, Log, TEXT("Downloading package: '%s'"), *Package.ToString());
+
+		const FString Args = FString::Printf(TEXT("package retrieve %s %s %s %s --unzip"),
+			*Package.Type, *Package.Name, *Package.Version, *Package.GetExecutableDir());
+
+		FString Result;
+		int32 ExitCode;
+		FSpatialGDKServicesModule::ExecuteAndReadOutput(*SpatialGDKServicesConstants::SpatialExe, Args,
+			*SpatialGDKServicesConstants::GDKProgramPath, Result, ExitCode);
+
+		if (ExitCode != 0)
+		{
+			UE_LOG(LogSpatialPackageService, Error, TEXT("Could not download package '%s' from package service"), *Package.ToString());
+			return false;
+		}
+
+		UE_LOG(LogSpatialPackageService, Log, TEXT("Downloaded package: %s"), *Result);
```

Comment:
`TEXT("Downloaded package '%s': %s"),*Package.ToString(), *Result);` ?

Reply:
The `Result` variable already contains the name among other stuff, IIRC